구현_실습/Boilerplate/src/js/StatusSection.js

```diff
@@ -0,0 +1,167 @@
+    divBtn.tabIndex = "0";
+    divBtn.addEventListener('keyup', (e) => {
+      if(e.key === "Enter") {
+        e.target.click();
+      }
+    })
+
+    return divBtn;
+  }
+
+  createActiveCntDiv() {
+    const cntSpan = createElement('span', StatusSection.activeCntClass);
+    cntSpan.innerText = '0';
+    this.activeCntSpan = cntSpan;
+    const cntTextSpan = document.createElement('span');
+    cntTextSpan.innerText = ' items left';
+    const cntDiv = document.createElement('div');
+    cntDiv.appendChild(cntSpan);
+    cntDiv.appendChild(cntTextSpan);
+
+    return cntDiv;
+  }
+
+  // 버튼 요소 생성
+  createFilterBtn(className, innerText) {
+    const btn = this.createDivBtnUsingTab(className);
+    btn.innerText = innerText;
+    btn.tabIndex = "0";
+    btn.addEventListener('click', (e) => {
+      if (e.target.innerText === this.status) {
+        return;
+      }
+      this.checkFilterBtn(e.target);
+      this.eventObserver?.changeStatusSection.notify(this.status);
+    });
+
+    return btn;
+  }
+
+  createAllBtn() {
+    return this.createFilterBtn(StatusSection.allBtnClass, StatusSection.STATUS.ALL);
+  }
+
+  createActiveBtn() {
+    return this.createFilterBtn(StatusSection.activeBtnClass, StatusSection.STATUS.ACTIVE);
+  }
+
+  createCompletedBtn() {
+    return this.createFilterBtn(StatusSection.completedBtnClass, StatusSection.STATUS.COMPLETED);
+  }
+
+  createClearCompletedBtn() {
+    const leftSpan = document.createElement('span');
+    leftSpan.innerText = 'Clear completed (';
+    const completedCntSpan = createElement('span', StatusSection.completedCntClass);
+    completedCntSpan.innerText = 0;
+    this.completedCntSpan = completedCntSpan;
+    const rightSpan = document.createElement('span');
+    rightSpan.innerText = ')';
+
+    const clearCompletedBtn = this.createDivBtnUsingTab(StatusSection.clearCompletedBtnClass);
+    clearCompletedBtn.addEventListener('click', () => {
+      if (this.completedCntSpan.innerText === '0') {
+        return;
+      }
+      this.completedCntSpan.innerText = 0;
+      this.eventObserver?.clearCompletedTodoItem.notify();
+    });
+    clearCompletedBtn.appendChild(leftSpan);
+    clearCompletedBtn.appendChild(completedCntSpan);
+    clearCompletedBtn.appendChild(rightSpan);
+
+    return clearCompletedBtn;
+  }
+
+  createEl() {
+    const allBtn = this.createAllBtn();
+    this.checkFilterBtn(allBtn);
+
+    const wrapperDiv = createElement('div', StatusSection.wrapperClass);
+    wrapperDiv.appendChild(this.createActiveCntDiv());
+    wrapperDiv.appendChild(allBtn);
+    wrapperDiv.appendChild(this.createActiveBtn());
+    wrapperDiv.appendChild(this.createCompletedBtn());
+    wrapperDiv.appendChild(this.createClearCompletedBtn());
+
+    this.wrapperDiv = wrapperDiv;
+  }
+
+  get() {
+    return this.wrapperDiv;
+  }
+
+  checkFilterBtn(btn) {
+    if (this.checkedItem) {
+      this.checkedItem.classList.remove(StatusSection.checkedFilterBtnClass);
+    }
```

Comment:
[옵셔널 체이닝](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Operators/Optional_chaining) 사용해도 좋을 것 같습니다.

```suggestion
    this.checkedItem?.classList.remove(StatusSection.checkedFilterBtnClass);
```